HeimdallrTests/Core/HeimdallrSpec.swift

```diff
@@ -21,6 +21,41 @@ class OAuthAccessTokenMockStore: OAuthAccessTokenStore {
     }
 }
 
+class OAuthAccessTokenInterceptorParser: OAuthAccessTokenParser {
+    
+    private let defaultParser = OAuthAccessTokenDefaultParser()
+    
+    var parseAccessTokenCalled: Bool = false
+    
+    private var shouldIntercept: Bool = false
+    private var interceptToken: OAuthAccessToken?
+    
+    func intercept(parsedToken token: OAuthAccessToken?) {
+        shouldIntercept = true
+        interceptToken = token
+    }
+    
+    func reset() {
+        parseAccessTokenCalled = false
+        shouldIntercept = false
+    }
+    
+    func parse(data: NSData) -> ParsedToken {
```

Comment:
I think it's better if we don't expose our internal dependency on Argo to users of the library. Seems a bit cleaner to me, but again, I can change it 😄